From 2ba61f8fbe9e799a7f4a2e6ebf0ac25a237e5222 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 16 Sep 2024 17:54:02 +0200
Subject: [PATCH 1218/1695] dts: stm32: stm32mp15: explicit GPIOs in non-secure
 state

Explicitly state that the GPIO controlling the SD card voltage
controller is non-secure, unless what the board panics with an error
message like:
E/TC:0 0 stm32_gpio_get_dt:779 node regulator-sd_switch requests secure GPIO F14 that cannot be secured

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 874d868e4b8d5ca12a2750e3ecb6933be1f5be4e dts: stm32: stm32mp15: explicit GPIOs in non-secure state]
Change-Id: Iad6c567241ea076dc82258c6ff91c777eb364481
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404742
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp157c-ed1.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 68362eb10..12aee5450 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -6,7 +6,7 @@
 /dts-v1/;
 
 #include <dt-bindings/clock/stm32mp1-clksrc.h>
-#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/gpio/stm32mp_gpio.h>
 #include <dt-bindings/mfd/st,stpmic1.h>
 #include <dt-bindings/power/stm32mp1-power.h>
 #include <dt-bindings/soc/stm32mp-provisioning.h>
@@ -85,7 +85,7 @@
 		regulator-type = "voltage";
 		regulator-always-on;
 
-		gpios = <&gpiof 14 GPIO_ACTIVE_HIGH>;
+		gpios = <&gpiof 14 (GPIO_ACTIVE_HIGH | GPIO_STM32_NSEC)>;
 		states = <1800000 0x1>,
 			 <2900000 0x0>;
 	};
-- 
2.47.2

